frontend/src/app/choreography.service.ts

```diff
@@ -15,8 +22,32 @@ export class ChoreographyService {
   }
 
   getChoreographiesById(id: number): Observable<QueryOutput<Choreography>> {
-    return query('choreography', id, (_id) =>
-      this.http.get<Choreography>(`/api/choreographies/${_id}`)
+    return query(
+      'choreography',
+      id,
+      (_id) => this.http.get<Choreography>(`/api/choreographies/${_id}`),
+      {
+        refetchOnWindowFocus: false,
+        refetchOnReconnect: false,
+      }
     );
   }
+
+  updateChoreography(choreography: Choreography): void {
+    const dialogRef = this.dialog.open(SaveChoreographyDialogComponent, {});
+    dialogRef
+      .afterClosed()
+      .pipe(
+        filter((result) => result === true),
+        switchMap(() => {
+          return this.http.put<Choreography>(
+            `/api/choreographies`,
```

Comment:
**suggestion:** add the id in the path even though we don't use it right now as it is the way to do RESTful interfaces